src/shogun/base/SGObject.h

```diff
@@ -1120,5 +1128,161 @@ CSGObject* get_by_tag(const CSGObject* obj, const std::string& name,
 
 #endif //DOXYGEN_SHOULD_SKIP_THIS
 #endif //SWIG
+
+template <class T>
+class ObservedValueTemplated;
+
+/**
+ * Observed value which is emitted by algorithms.
+ */
+class ObservedValue : public CSGObject
+{
+public:
+	/**
+	 * Constructor
+	 * @param step step
+	 * @param name name of the observed value
+	 */
+	ObservedValue(int64_t step, std::string name);
+
+	/**
+	 * Destructor
+	 */
+	~ObservedValue(){};
+
+#ifndef SWIG
+	/**
+	* Return a any version of the stored type.
+	* @return the any value.
+	*/
+	virtual Any& get_any()
```

Comment:
what happens in swig?

Reply:
It’s simply not needed there

Reply:
so how do i get the observed value in swig?

Reply:
Get

Reply:
That’s the whole point of this pr

Reply:
this was already true prior pr, as observevalue is sgobject-like already... alas it wasn't registered... but imo it's really an unnecessary overhead to have this whole thing sgobject-like for the sake of registering and having the getter interface

Reply:
should have said series of pr. I agree I dont like the overhead as well. I would like to carve that functionality out of SGObject into a base class that just serves the parameter registration/access purposes. But this is of course out of scope for this little PR here.
The getter interface is a huge advantage as otherwise we once again need to expose/implement/swig-expose a lot of types. Let's discuss this next time on video